lib/puma/server.rb

```diff
@@ -869,35 +869,32 @@ def graceful_shutdown
       end
     end
 
-    # Stops the acceptor thread and then causes the worker threads to finish
-    # off the request queue before finally exiting.
-    #
-    def stop(sync=false)
+    def notify_safely(message)
       begin
-        @notify << STOP_COMMAND
+        @notify << message
       rescue IOError
         # The server, in another thread, is shutting down
+      rescue RuntimeError => e
+        raise e unless e.message.include?('IOError')
```

Comment:
Can you leave a comment here explaining this? Otherwise it looks pretty magical.